lib/diagnostics_channel.js

```diff
@@ -105,7 +108,10 @@ class ActiveChannel {
     const index = ArrayPrototypeIndexOf(this._subscribers, subscription);
     if (index === -1) return false;
 
-    ArrayPrototypeSplice(this._subscribers, index, 1);
+    const before = ArrayPrototypeSlice(this._subscribers, 0, index);
+    const after = ArrayPrototypeSlice(this._subscribers, index + 1);
+    this._subscribers = before;
+    ArrayPrototypePushApply(this._subscribers, after);
```

Comment:
I use `ArrayPrototypePushApply()` instead of `ArrayPrototypeConcat()` because according to this PR: https://github.com/nodejs/node/pull/44445 it's not recommended to use concat.